### PR TITLE
Use OCR validator

### PR DESCRIPTION
**Pull request by rkondratowicz**
created in my previous PR

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java [hunk 1]
```diff
@@ -55,6 +56,7 @@ public FailedDocUploadProcessor(
         EnvelopeRepository envelopeRepository,
         ProcessEventRepository eventRepository,
         ContainerMappings containerMappings, // NOSONAR
+        OcrValidator ocrValidator, // NOSONAR
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java]: inheritance 😢

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java]: what does it say?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java]: Note that this job (faied doc upload) does not need this validator at all, but it's injected for the compatibility reasons.
Same as `ContainerMappings` and `ServiceBusHelper`

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java [hunk 2]
```diff
@@ -101,6 +102,9 @@
     @Mock
     private DocumentManagementService documentManagementService;
 
+    @Mock
+    private OcrValidator ocrValidator;
```
Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: Given OCR is a part of the metadata file, I wonder if OCR validator should be a dependency of envelope validator.

Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: all methods in EnvelopeValidator are currently static :/

Comment (lgonczar) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: One reason to not make static tools, to remember.

**Pull request merged**